docs/posts/ibis-duckdb-geospatial/index.qmd

```diff
@@ -0,0 +1,242 @@
+con.list_tables()
+```
+
+We have multiple tables with information about New York City. Following Dr. Wu's class, we'll take a look at some
+spatial relations.
+
+We can start by taking a peak at the `nyc_subway_stations` table.
+
+```{python}
+subway_stations = con.table('nyc_subway_stations')
+subway_stations
+```
+
+Notice that the last column has a `geometry` type, and in this case it contains points that represent the location of
+each subway station. Let's grab the entry for `'Broad St'`.
+
+```{python}
+broad_station = subway_stations.filter(subway_stations.NAME == "Broad St")
+broad_station
+```
+
+### `geo_equals` (`ST_Equals`)
+
+In DuckDB `ST_Equals` returns `True` if two geometries are topologically equal. This means that they have the same
+dimension and identical coordinate values, although the order of the vertices may be different.
+
+The following is a bit redundant but we can check if our `"Broad St"` point matches only one point in our data using
+`geo_equals`
+
+```{python}
+subway_stations.filter(subway_stations.geom.geo_equals(broad_station.geom))
+```
+
+We can also write this query without using `broad_station` as a variable, and with the help of the deferred expressions
+API, also known as [the underscore API](https://ibis-project.org/how-to/analytics/chain_expressions.html).
+
+```{python}
+subway_stations.filter(_.geom.geo_equals(_.filter(_.NAME == "Broad St").geom))
+```
+
+### `intersect` (ST_Intersect)
+
+Let's find in which neighborhood is "Broad Street" subway station and determine its neighborhood using the the
+geospatial `intersect` function. The `intersect` function returns `True` if two geometries have any points in common.
+
+```{python}
+boroughs = con.table("nyc_neighborhoods")
+boroughs
+```
+
+```{python}
+boroughs.filter(boroughs.geom.intersects(broad_station.select(broad_station.geom).to_array()))
+```
+
+### `d_within` (ST_DWithin)
+
+We can also find what are the streets nearby (within 10 meters of) of the "Broad St" subway station using the `d_within`
+function. The `d_within` function returns True if the geometries are within a given distance.
+
+```{python}
+streets = con.table("nyc_streets")
+streets
+```
+
+Using the deferred API, we can check which streets are within `d=10` meters of distance.
+
+```{python}
+sts_near_broad =streets.filter(_.geom.d_within(broad_station.select(_.geom).to_array(), 10))
+sts_near_broad
+```
+
+::: {.callout-note}
+In the previous query, `streets` and `broad_station` are different tables. We use [`to_array()`](https://ibis-project.org/reference/expression-tables.html#ibis.expr.types.relations.Table.to_array) to generate a
+scalar subquery from a table with a single column (whose shape is scalar).
+:::
+
+To visualize the findings, we will convert the tables to Geopandas dataframes. We need to do this because,
+at the moment, DuckDB geometries are not capable to store SRID information. By converting to
+Geopandas we can assign a `crs` and make some nice plots.
```

Comment:
```suggestion
To visualize the findings, we will convert the tables to Geopandas DataFrames.
```

I think this is required regardless of whether DuckDB supports storing SRID information.